### PR TITLE
[DB-14414] Filtering queries based on tables schema in the query for unsupported query constructs

### PR DESCRIPTION
**Pull request by sanyamsinghal**
### Describe the changes in this pull request
- Removed collection of functions names in the queries being detected for unsupported query constructs.
- Only considering the table objects(schema names of them) to compare with the user provided schema list.

### Describe if there are any user-facing changes

N/A, only the queries to be displayed in UQC assessment report would change based on the schema list provided by the user.

### How was this pull request tested?

- Added unit test to verify that ObjectCollector is collecting all objects and table objects, depending on the predicate provided to it.

- Added a new end to end test `pg/assessment-report-test-uqc`
- Also manually tested end to end scenario, where HTML report doesn't show the queries from different schema tables.

### Does your PR have changes that can cause upgrade issues? 
| Component        | Breaking changes? |
| :----------------------------------------------: | :-----------: |
| MetaDB                                                    | No |
| Name registry json                                        |  No |
| Data File Descriptor Json                                 | No |
| Export Snapshot Status Json                               | No |
| Import Data State                                         | No |
| Export Status Json                                        | No |
| Data .sql files of tables                                 | No |
| Export and import data queue                              | No |
| Schema Dump                                               | No |
| AssessmentDB                                              | No |
| Sizing DB                                                 | No |
| Migration Assessment Report Json                          | No |
| Callhome Json                                             | No |
| YugabyteD Tables                                          | No |
| TargetDB Metadata Tables                                  | No |

### CODE REVIEW

File: yb-voyager/src/query/queryissue/detectors_test.go [hunk 1]
```diff
@@ -627,7 +627,7 @@ func TestCombinationOfDetectors1WithObjectCollector(t *testing.T) {
 		visited := make(map[protoreflect.Message]bool)
 		unsupportedConstructs := []string{}
 
-		objectCollector := queryparser.NewObjectCollector()
+		objectCollector := queryparser.NewObjectCollector(nil)
```
Comment (makalaaneesh) [inline yb-voyager/src/query/queryissue/detectors_test.go:630]: TODO?

File: yb-voyager/src/query/queryparser/traversal_test.go [hunk 2]
```diff
@@ -237,9 +239,117 @@ func TestObjectCollector2(t *testing.T) {
...
+			ExpectedObjects: []string{"pg_depend"},
+			ExpectedSchemas: []string{""},
+		},
+		{
+			Sql:             `SELECT pg_advisory_unlock_shared(100);`,
+			ExpectedObjects: []string{}, // empty slice represents no object collected
+			ExpectedSchemas: []string{},
+		},
+		{
+			Sql:             `SELECT xpath_exists('/employee/name', '<employee><name>John</name></employee>'::xml)`,
```
Comment (makalaaneesh) [inline yb-voyager/src/query/queryparser/traversal_test.go:326]: can you add a test that has
-  pg_advisory_lock
-  xml_functions
- function call of another schema y
- table of schema x

Assert that objects returned are x.table, schemas returned are x.

File: yb-voyager/src/query/queryparser/object_collector.go [hunk 3]
```diff
@@ -56,7 +77,10 @@ func (c *ObjectCollector) Collect(msg protoreflect.Message) {
 		relName := GetStringField(msg, "relname")
 		objectName := lo.Ternary(schemaName != "", schemaName+"."+relName, relName)
 		log.Debugf("[RangeVar] fetched schemaname=%s relname=%s objectname=%s field\n", schemaName, relName, objectName)
-		c.addObject(objectName)
+		// it will be either table or view, considering objectType=table for both
+		if c.predicate(schemaName, relName, "table") {
```
Comment (makalaaneesh) [inline yb-voyager/src/query/queryparser/object_collector.go:81]: use constants for table, function

File: yb-voyager/src/query/queryparser/object_collector.go [hunk 4]
```diff
@@ -23,14 +23,35 @@ import (
...
+
+// AllObjectsPredicate always returns true, meaning it will collect all objects found.
+func AllObjectsPredicate(schemaName, objectName, objectType string) bool {
+	return true
 }
 
-func NewObjectCollector() *ObjectCollector {
+// TablesOnlyPredicate returns true only if the object type is "table".
+func TablesOnlyPredicate(schemaName, objectName, objectType string) bool {
+	return objectType == "table"
```
Comment (makalaaneesh) [inline yb-voyager/src/query/queryparser/object_collector.go:44]: use constant for table

File: yb-voyager/src/query/queryparser/object_collector.go [hunk 5]
```diff
@@ -23,14 +23,35 @@ import (
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
-// ObjectCollector collects unique schema-qualified object names.
+// ObjectPredicate defines a function signature that decides whether to include an object.
+type ObjectPredicate func(schemaName string, objectName string, objectType string) bool
```
Comment (makalaaneesh) [inline yb-voyager/src/query/queryparser/object_collector.go:27]: Looks neat :)

File: yb-voyager/src/query/queryparser/traversal_test.go (hunk 2 above)
Comment (sanyamsinghal) [reply yb-voyager/src/query/queryparser/traversal_test.go:326]: done

**Review by makalaaneesh — APPROVED** [review]
LGTM

**Comment by sanyamsinghal** [comment]
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/4251/

**Pull request merged**